src/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/WrapGh.hpp

```diff
@@ -0,0 +1,191 @@
+#include "Options/Options.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+// IWYU pragma: no_forward_declare Tags::deriv
+
+/// \cond
+class DataVector;
+namespace PUP {
+class er;
+}  // namespace PUP
+/// \endcond
+
+namespace GeneralizedHarmonic {
+namespace Solutions {
+
+/*!
+ * \brief A wrapper for general-relativity analytic solutions that loads
+ * the analytic solution and then adds a function that returns
+ * any combination of the generalized-harmonic evolution variables,
+ * specifically `gr::Tags::SpacetimeMetric`, `GeneralizedHarmonic::Tags::Pi`,
+ * and `GeneralizedHarmonic::Tags::Phi`
+ */
+template <typename SolutionType>
+class WrapGh : public SolutionType {
+ public:
+  using SolutionType::SolutionType;
+
+  static constexpr size_t volume_dim = SolutionType::volume_dim;
+  using options = typename SolutionType::options;
+  static constexpr OptionString help = SolutionType::help;
+
+  using DerivLapse = ::Tags::deriv<gr::Tags::Lapse<DataVector>,
+                                   tmpl::size_t<volume_dim>, Frame::Inertial>;
+  using DerivShift =
+      ::Tags::deriv<gr::Tags::Shift<volume_dim, Frame::Inertial, DataVector>,
+                    tmpl::size_t<volume_dim>, Frame::Inertial>;
+  using DerivSpatialMetric = ::Tags::deriv<
+      gr::Tags::SpatialMetric<volume_dim, Frame::Inertial, DataVector>,
+      tmpl::size_t<volume_dim>, Frame::Inertial>;
+  using TimeDerivLapse = ::Tags::dt<gr::Tags::Lapse<DataVector>>;
+  using TimeDerivShift =
+      ::Tags::dt<gr::Tags::Shift<volume_dim, Frame::Inertial, DataVector>>;
+  using TimeDerivSpatialMetric = ::Tags::dt<
+      gr::Tags::SpatialMetric<volume_dim, Frame::Inertial, DataVector>>;
+
+  using IntermediateVars = tuples::tagged_tuple_from_typelist<
+      typename SolutionType::template tags<DataVector>>;
+
+  template <typename... Tags>
+  tuples::TaggedTuple<Tags...> variables(
+      const tnsr::I<DataVector, volume_dim>& x, double t,
+      tmpl::list<Tags...> /*meta*/) const noexcept {
+    // Get the underlying solution's variables using the solution's tags list,
+    // store in IntermediateVariables
+    const IntermediateVars& intermediate_vars = SolutionType::variables(
+        x, t, typename SolutionType::template tags<DataVector>{});
+
+    return {
+        get<Tags>(variables(x, t, tmpl::list<Tags>{}, intermediate_vars))...};
+  }
+
+  template <typename Tag>
+  tuples::TaggedTuple<Tag> variables(const tnsr::I<DataVector, volume_dim>& x,
+                                     double t, tmpl::list<Tag> /*meta*/) const
+      noexcept {
+    const IntermediateVars& intermediate_vars = SolutionType::variables(
+        x, t, typename SolutionType::template tags<DataVector>{});
+    return {get<Tag>(variables(x, t, tmpl::list<Tag>{}, intermediate_vars))};
+  }
+
+ private:
+  tuples::TaggedTuple<gr::Tags::Lapse<DataVector>> variables(
```

Comment:
can't pretty much all of these except the GH variables just be inherited? I don't think you'd even need to do `using SolutionType::variables`, but maybe.

Reply:
Discussed with @nilsdeppe (thanks!) ... I tried this but it didn't compile, and @nilsdeppe is looking to see if there's a nicer solution than just leaving this as is

Reply:
Here:

```cpp
  using TagShift = gr::Tags::Shift<volume_dim, Frame::Inertial, DataVector>;
  using TagSpatialMetric =
      gr::Tags::SpatialMetric<volume_dim, Frame::Inertial, DataVector>;
  using TagInverseSpatialMetric =
      gr::Tags::InverseSpatialMetric<volume_dim, Frame::Inertial, DataVector>;
  using TagExCurvature =
      gr::Tags::ExtrinsicCurvature<volume_dim, Frame::Inertial, DataVector>;

#define FUNC_DECL(r, data, elem)                                            \
  tuples::TaggedTuple<elem> variables(                                      \
      const tnsr::I<DataVector, volume_dim>& /*x*/, double /*t*/,           \
      tmpl::list<elem> /*meta*/, const IntermediateVars& intermediate_vars) \
      const noexcept;

#define MY_LIST                                                              \
  BOOST_PP_TUPLE_TO_LIST(                                                    \
      11,                                                                    \
      (gr::Tags::Lapse<DataVector>, TimeDerivLapse, DerivLapse, TagShift,    \
       TimeDerivShift, DerivShift, TagSpatialMetric, TimeDerivSpatialMetric, \
       DerivSpatialMetric, TagInverseSpatialMetric, TagExCurvature,          \
       gr::Tags::SqrtDetSpatialMetric<DataVector>))

  BOOST_PP_LIST_FOR_EACH(FUNC_DECL, _, MY_LIST)
#undef MY_LIST
#undef FUNC_DECL
```
and you need the includes:
```cpp
#include <boost/preprocessor/list/for_each.hpp>
#include <boost/preprocessor/tuple/to_list.hpp>
```